hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/clustering/run/strategy/MultipleSparkJobExecutionStrategy.java

```diff
@@ -575,7 +575,7 @@ public Iterator<InternalRow> call(ClusteringOperation clusteringOperation) throw
             getHoodieTable().getMetaClient(),
             getHoodieTable().getMetaClient().getTableConfig().getProps(),
             0,
-            Long.MAX_VALUE,
+            -1,
```

Comment:
can we add java doc here as to why we are setting it to -1 ?

Reply:
on second thought, why can't we deduce base file or bootstrap file and deduce file lenght here only and keep the file group reader simple ?

Reply:
We want to avoid file system calls on getting the status of files as much as possible.  In most cases, the file system view already provides the file information including the length.  Since the compaction and clustering plans do not store the file length information, we have to pass in `-1` here for the file group reader to find the length in an extra file system call.

Reply:
Added docs